docs/src/pages/components/checkbox.mdx

```diff
@@ -0,0 +1,166 @@
+---
+name: Checkbox
+route: /components/checkbox
+---
+
+import { Intro, Types, Usage } from '../../components/component'
+
+# Checkbox
+
+## Examples
+### Checked
+Passing the ```checked``` indicates that the checkbox is selected.
+
+```typescript
+import React from 'react'
+import Checkbox from '@pluralsight/ps-design-system-checkbox'
+
+const Comp = () => (
+  <div className="example-flex-column">
+    <Checkbox checked label="I'm checked" value="Checked-eg" />
+    <Checkbox label="I'm not" value="Checked-eg" />
+  </div>
+)
+
+export default Comp
+
+```
+### Indeterminate
+Passing the ```indeterminate``` flag indicates that the checkbox selection should be obscured as if the control was in a third, indeterminate, state.
+
+```typescript
+import React from 'react'
+import Checkbox from '@pluralsight/ps-design-system-checkbox'
+
+const Comp = () => (
+  <div className="example-flex-column">
+    <Checkbox indeterminate label="I'm indeterminate" value="Indeterminate-eg"/>
+    <Checkbox checked indeterminate label="I'm checked but still indeterminate" value="Indeterminate-eg"/>
+  </div>
+)
+
+export default Comp
+
+```
+
+### Disabled
+Disabled checkboxes are unmodifiable and diminished visually.
+
+```typescript
+import React from 'react'
+import Checkbox from '@pluralsight/ps-design-system-checkbox'
+
+const Comp = () => (
+  <div className="example-flex-column">
+    <Checkbox disabled label="Can't touch this" value="Disabled-eg"/>
+    <Checkbox disabled checked label="Checked or not" value="Disabled-eg"/>
+  </div>
+)
+
+export default Comp
+
+```
+
+### Error
+description: Error states are engaged with the ```error``` flag.
+
+```typescript
+import React from 'react'
+import Checkbox from '@pluralsight/ps-design-system-checkbox'
+
+const Comp = () => (
+  <div className="example-flex-column">
+    <Checkbox error label="I've got problems"  value="Error-eg"/>
+    <Checkbox error checked label="Checked and still not good enough" value="Error-eg"/>
+  </div>
+)
+
+export default Comp
+
+```
+
+### In-app
+
+```typescript
+import React from 'react'
+import Checkbox from '@pluralsight/ps-design-system-checkbox'
+import { layout, type, colorsTextIcon, colorsBackgroundLight } from '@pluralsight/ps-design-system-core'
+
+class Comp extends React.Component {
```

Comment:
Could think about doing a functional component rewrite of this to be more modern.

Reply:
Tried that got annoyed when I realized it wasn't a simple 1-1 refactor.  Decided I rather refactor checkbox to work  more like radio in separate PR. Not going to change this one yet as it's outside the scope of the issue.